### ISSUE TITLE
Server component

### ISSUE THREAD

**Issue by chrismytton** [everypolitician/membership_matrix#9]
Now we've got a UI we need to be able to push updates into EveryPolitician.

So we need to create a server component that handles authenticating and authorizing the user. When a user edits some information about a country, we need to somehow turn that into a Pull Request on EveryPolitician. It would be nice if this repo could remain a simple static site. Then the user would need to go off to a dynamic endpoint, authorize with it, then they'd be redirected back to the static site. Then when they came to saving the data we'd use the fact that the user has a session established with the dynamic endpoint to POST the data back there, and then the endpoint can handle the details of pushing it into EveryPolitician.

Simply opening a pull request containing a new copy of the Popolo isn't going to be enough, since the Popolo gets rebuilt from the source data each time. I think we'll need to explicitly add the membership_matrix as a source for data, like we've done with Gender Balance. Then use that data to regenerate the Popolo file. Can we just get this to place a CSV file at a known location in the `sources` directory for a legislature and then read from that? If we do that then how can we indicate that a membership in the CSV file is overriding an existing membership?

**Comment by tmtmtmtm** [comment]
On the first point: I'm not sure that we actually _need_ authentication/authorisation. It would certainly be useful to know who's triggering the change in lots of cases, but if all we're doing is opening a PR I'm not so sure that that's essential. (And for specific partner sites we probably want to find a way to integrate this into some other workflow, e.g. by being able to display this within a pre-auth-ed Pombola admin).

On the second — yep, dropping this in as new source of 'membership' information at a known location seems like a good approach. We'd either need to add something to the instructions file to tell it about that file if we're doing it for a handful of countries, or it could be a file that we just presume is at a known location (like we [currently do for corrections files](https://github.com/everypolitician/everypolitician-data/blob/9d8a219caba79b1e6e2faf7490e9a41b66edc2c1/rake_helpers/combine_sources.rb#L276-L278))

**Comment by chrismytton** [comment]
@tmtmtmtm On the auth point I had thought it would be nice to have people sign in with GitHub, then we could use their credentials to make the pull request come from their account, which would make it more obvious that it wasn't just an automated update from a bot. It would also have the nice bonus that they would become a "contributor" on the everypolitician-data repo. Having said that I agree that it isn't necessary for a first pass so we can probably leave that as a nice to have in the future.

**Comment by tmtmtmtm** [comment]
Yep — it would definitely be _nice_ to have, but keeping it out of the critical path of the first version is probably sensible.
